src/c.c

```diff
@@ -61,6 +63,12 @@ r_obj* vec_c_opts(r_obj* xs,
     ptype = KEEP(vec_ptype_common_opts(xs, ptype, &ptype_opts));
   }
 
+  struct cast_common_opts cast_opts = {
+    .call = error_call,
+    .fallback = *fallback_opts
+  };
+  xs = KEEP(vec_cast_common_opts(xs, ptype, &cast_opts));
```

Comment:
This does mean that we allocate another list the same size as `xs` which we didn't do before, but that seems worth it to let the `vec_cast_common_opts()` loop handle all the details of getting the positional errors right